src/engine/types/actions.rs

```diff
@@ -47,6 +50,15 @@ impl<T> EngineAction for CreateAction<T> {
     }
 }
 
+impl fmt::Display for CreateAction<PoolUuid> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            CreateAction::Created(uuid) => write!(f, "uuid {} created", uuid.to_simple_ref()),
```

Comment:
Note that here you know that this is a CreateAction for a Pool, because the type you're implementing ```Display``` for is ```CreateAction<PoolUuid>```. So, just here it is reasonable to include that information in the string for ```Created``` just as you already do in the string for ```Identity```. That is, in both places it makes sense to state that this is a _pool_ that is being discussed.

Reply:
I think I figured out what's going on: for the CreateAction, the caller of the macro provides the pool name:
```
let result = pool_op_logging!("preparing to create pool with name {}", name; "pool with name {}, {}", name; engine.create_pool(name, &blockdevs, tuple_to_option(redundancy)));
```
...and for RenameAction, the caller of the macro provides the new name:
```
let result = pool_op_logging!("preparing to rename pool with name {}, uuid {}", new_name, pool_uuid.to_simple_ref(); "pool with name {}, {}", new_name; engine.rename_pool(pool_uuid, new_name));
```

(I need a little help with line continuations in the above two lines; it's probably making things more difficult than they should be.)

The macro callers provide the format strings to use, but I think I was only able at first to add the string after what the action prints.

Reply:
Here's the big challenge: currently the action only handles the uuid.  However, some callers of the macro have the desired name available.

Do I abandon the printing of the pool name?  Or is there a way to also print the name in the action?  Or is there some other way to store the "pool with..." string in the action without running into syntax problems?

Reply:
There are really two choices:

*  You can only use the data stored in the action values or its type, like the fact that it is about the PoolUuid. So, if the name isn't available, then the only thing to do is actually to change the structure of the action, which might not be desirable.
* The alternative is to work out a format string and arguments passed in the macro expansion that would do the trick. That's probably the less heavyweight approach.

Reply:
I tried standardizing the action messages with the last commit, and here's the result:
```
 INFO libstratis::dbus_api::api::manager_2_0::methods: preparing to create pool with name spool1
 INFO libstratis::engine::strat_engine::thinpool::thinpool: Attempting to extend thinpool data sub-device belonging to pool 16887092b46d4b35acc564d79c541f2c by 15095808 sectors
 INFO libstratis::engine::strat_engine::thinpool::thinpool: Extended thinpool data sub-device belonging to pool with uuid 16887092b46d4b35acc564d79c541f2c by 15095808 sectors
 INFO libstratis::dbus_api::api::manager_2_0::methods: pool with name spool1, pool uuid 16887092b46d4b35acc564d79c541f2c created

 INFO libstratis::dbus_api::pool::pool_2_0::methods: preparing to rename pool with name spool2, uuid 16887092b46d4b35acc564d79c541f2c
 INFO libstratis::dbus_api::pool::pool_2_0::methods: pool with name spool2, pool uuid 16887092b46d4b35acc564d79c541f2c renamed

 INFO libstratis::dbus_api::api::manager_2_0::methods: preparing to destroy pool with uuid 16887092b46d4b35acc564d79c541f2c
 INFO libstratis::dbus_api::api::manager_2_0::methods: pool uuid 16887092b46d4b35acc564d79c541f2c destroyed
```

Reply:
> I tried standardizing the action messages with the last commit, and here's the result:
> 
> ```
>  INFO libstratis::dbus_api::api::manager_2_0::methods: preparing to create pool with name spool1
>  INFO libstratis::engine::strat_engine::thinpool::thinpool: Attempting to extend thinpool data sub-device belonging to pool 16887092b46d4b35acc564d79c541f2c by 15095808 sectors
>  INFO libstratis::engine::strat_engine::thinpool::thinpool: Extended thinpool data sub-device belonging to pool with uuid 16887092b46d4b35acc564d79c541f2c by 15095808 sectors
>  INFO libstratis::dbus_api::api::manager_2_0::methods: pool with name spool1, pool uuid 16887092b46d4b35acc564d79c541f2c created
> 
>  INFO libstratis::dbus_api::pool::pool_2_0::methods: preparing to rename pool with name spool2, uuid 16887092b46d4b35acc564d79c541f2c
>  INFO libstratis::dbus_api::pool::pool_2_0::methods: pool with name spool2, pool uuid 16887092b46d4b35acc564d79c541f2c renamed
> 
>  INFO libstratis::dbus_api::api::manager_2_0::methods: preparing to destroy pool with uuid 16887092b46d4b35acc564d79c541f2c
>  INFO libstratis::dbus_api::api::manager_2_0::methods: pool uuid 16887092b46d4b35acc564d79c541f2c destroyed
> ```

This is looking better and better! I think for destroy pool, though, you're missing a "with". Also, although I'm not a stickler for identifier names, for log entries i prefer UUID in all caps.

Reply:
Ah yes, the missing "with" on destroy.  Fixing this will make it "stand out" in actions.rs, but if we can find a way to have the destroy caller add the name as a preamble (as create and rename do), then this can be straightened out later.